jax/_src/lru_cache.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2024 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import logging
+import os
+import tensorflow as tf
```

Comment:
Instead of using tf directly, can you use jax._src.path.Path like gfile_cache.py does? This way we don't take a hard dependency on tensorflow.

https://github.com/google/jax/blob/main/jax/_src/path.py uses [etils](https://github.com/google/etils) if available, which does pull in TF to support GCS, but falls back to [pathutil](https://docs.python.org/3/library/pathlib.html) if not, meaning you only have access to the local filesystem.

Reply:
Done